sklearn/linear_model/ridge.py

```diff
@@ -367,11 +367,27 @@ def _ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
                       return_n_iter=False, return_intercept=False,
                       X_scale=None, X_offset=None):
 
-    if return_intercept and sparse.issparse(X) and solver != 'sag':
-        if solver != 'auto':
-            warnings.warn("In Ridge, only 'sag' solver can currently fit the "
-                          "intercept when X is sparse. Solver has been "
-                          "automatically changed into 'sag'.")
+    has_sw = sample_weight is not None
+
+    if solver == 'auto':
+        if return_intercept:
+            # only sag supports fitting intercept directly
+            solver = "sag"
+        elif has_sw:
+            # this should be changed since all solvers support sample_weights
+            solver = "cholesky"
+        elif sparse.issparse(X):
+            solver = "sparse_cg"
+        else:
+            solver = "cholesky"
+
+    if solver not in ('sparse_cg', 'cholesky', 'svd', 'lsqr', 'sag', 'saga'):
+        raise ValueError('Solver %s not understood' % solver)
```

Comment:
`'Solver %s not understood' % solver`
->
`"Known solver are  'sparse_cg', 'cholesky', 'svd', 'lsqr', 'sag' or 'saga'. Got %s." % solver`

Reply:
done